core/src/cpu/air/mod.rs

```diff
@@ -420,6 +434,53 @@ impl CpuChip {
             .assert_eq(local.pc, next.pc);
     }
 
+    /// Constraints related to the commit instruction.
+    pub(crate) fn commit_eval<AB: SP1AirBuilder>(
+        &self,
+        builder: &mut AB,
+        local: &CpuCols<AB::Var>,
+        is_commit: AB::Expr,
+    ) {
+        let public_values = builder.public_values();
+
+        // Convert public values into words.
+        let mut digest_words = Vec::new();
+        for bytes in public_values.chunks_exact(WORD_SIZE) {
+            let bytes_expr_vec: Vec<AB::Expr> =
+                bytes.iter().map(|byte| (*byte).into()).collect::<Vec<_>>();
+
+            digest_words.push(Word::<AB::Expr>(bytes_expr_vec.try_into().unwrap()));
+        }
+
+        // Get the ecall specific columns.
+        let ecall_columns = local.opcode_specific_columns.ecall();
+
+        // Verify the index bitmap.
+        let mut bitmap_sum = AB::Expr::zero();
```

Comment:
nit: it's kind of weird here that you do pv stuff -> ecall/index bitmap stuff -> pv stuff. it should be 1) pv stuff and then 2) ecall/index bitmap stuff or vice-versa.